src/momento/aio/_add_header_client_interceptor.py

```diff
@@ -20,6 +34,11 @@ async def intercept_unary_unary(
     ) -> Union[grpc.aio._call.UnaryUnaryCall, grpc.aio._typing.ResponseType]:
         if client_call_details.metadata is None:
             client_call_details.metadata = Metadata()
-        client_call_details.metadata.add(self._header_name, self._header_value)
+        for header in self.headers_to_add_every_time:
+            client_call_details.metadata.add(header.name, header.value)
+        if AddHeaderClientInterceptor.are_only_once_headers_sent == False:
+            for header in self._headers_to_add_once:
```

Comment:
this seems much safer and cleaner then the `[0]` reference before 👍

Reply:
Yup, as @gautamomento in the csharp sdk pr, I'm trying to implement future proof option rather than fixed index here.